### PR TITLE
feat: Initial for packagers

### PR DESCRIPTION
**Pull request by djpolygon**
## Description

This is an initial for enabling debian and rpm packaging for arm and amd64/x86 for cdk.

Once approved, I will enable the arm runner( name may change, so expecting to adjust the label for the arm specific builders).

### CODE REVIEW

**Review by sshrihar — APPROVED** [review]

File: .github/workflows/arm_deb_packager.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          ARCH: arm64
+
+      - name: Release cdk Packages
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ env.GIT_TAG }}
+          prerelease: true
+          files: |
+            packaging/deb/cdk**.deb
+            packaging/deb/cdk**.deb.checksum
```
Comment (vcastellm) [inline .github/workflows/arm_deb_packager.yml:87]: nit: add new line

File: .github/workflows/x86_deb_packager.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+
+      - name: Release cdk Packages
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ env.GIT_TAG }}
+          prerelease: true
+          files: |
+            packaging/deb/cdk**.deb
+            packaging/deb/cdk**.deb.checksum
```
Comment (vcastellm) [inline .github/workflows/x86_deb_packager.yml:87]: nit: add new line

File: packaging/deb/cdk/DEBIAN/postinst [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#
+adduser --disabled-password --disabled-login --shell /usr/sbin/nologin --quiet --system --no-create-home --home /nonexistent cdk
+if [ -d "/opt/cdk" ]
+then
+    echo "Directory /opt/cdk exists."
+else
+     mkdir -p /opt/cdk
+     chown -R cdk /opt/cdk
+fi
+systemctl daemon-reload
```
Comment (vcastellm) [inline packaging/deb/cdk/DEBIAN/postinst:12]: nit: add new line

File: packaging/deb/cdk/DEBIAN/postrm [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+#
+###############
+# Remove cdk-node installs
+##############
+#rm -rf /lib/systemd/system/cdk-node.service
+deluser cdk
+#systemctl daemon-reload
```
Comment (vcastellm) [inline packaging/deb/cdk/DEBIAN/postrm:8]: nit: add new line

File: packaging/systemd/cdk-node.service.example [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  Restart=on-failure
+  RestartSec=5s
+  ExecStart=/usr/bin/cdk-node -c $config
+  Type=simple
+  KillSignal=SIGINT
+  User=cdk
+  TimeoutStopSec=120
+
+[Install]
+  WantedBy=multi-user.target
```
Comment (vcastellm) [inline packaging/systemd/cdk-node.service.example:16]: nit: add new line

File: packaging/systemd/cdk-node.service.example [hunk 6]
```diff
@@ -0,0 +1,16 @@
+[Unit]
+  Description=cdk
+  StartLimitIntervalSec=500
+  StartLimitBurst=5
+
+[Service]
+  Restart=on-failure
+  RestartSec=5s
+  ExecStart=/usr/bin/cdk-node -c $config
```
Comment (vcastellm) [inline packaging/systemd/cdk-node.service.example:9]:
```suggestion
  ExecStart=/usr/bin/cdk --config $config --chain $chain node
```

**Comment by vcastellm** [comment]
@djpolygon plz, merge master in order to fix e2e tests

**Comment by djpolygon** [comment]
@vcastellm the branch was develop and merged

File: .github/workflows/arm_deb_packager.yml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        run: echo "GIT_TAG=`echo $(git describe --tags --abbrev=0)`" >> $GITHUB_ENV
+      - name: adding version
+        run: |
+          NUMERIC_VERSION=$( echo ${{ env.GIT_TAG }} | sed 's/[^0-9.]//g' ) 
+          echo "VERSION=$NUMERIC_VERSION" >> $GITHUB_ENV
+
+      - name: go mod download
+        run: go mod download
+
+      - name: build the binary
```
Comment (Stefan-Ethernal) [inline .github/workflows/arm_deb_packager.yml:41]:
```suggestion
      - name: Build the binary
```

File: packaging/systemd/cdk.service.example [hunk 8]
```diff
@@ -0,0 +1,16 @@
+[Unit]
+  Description=cdk
+  StartLimitIntervalSec=500
+  StartLimitBurst=5
+
+[Service]
+  Restart=on-failure
+  RestartSec=5s
+  ExecStart=/usr/bin/cdk --config $config --chain $chain node
```
Comment (joanestebanr) [inline packaging/systemd/cdk.service.example:9]: This params are wrong:
- `--config` : may be is `--cfg`
- `--chain`:  no idea
Also is missing the list oc components to run: `--components`

The path for binary is also wrong

Comment (djpolygon) [reply packaging/systemd/cdk.service.example:9]: Yes this was only an example, this is actually never used currently. I had discussed this with @vcastellm.

File: .github/workflows/arm_deb_packager.yml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        run: make build
+
+      - name: build the rust binary
+        run: cargo build --release --bin cdk
```
[the remark this answers is not in the document]
Comment (vcastellm) [reply .github/workflows/arm_deb_packager.yml]: I think it's good to maintain explicitly separated binary builds everywhere except in development, so we'll need here that to set `BUILD_SCRIPT_DISABLED=1` env var before building @djpolygon

File: packaging/systemd/cdk.service.example (hunk 8 above)
Comment (vcastellm) [reply packaging/systemd/cdk.service.example:9]: It's actually called `--config` in the cli, for the sake of consistency

File: .github/workflows/arm_deb_packager.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (djpolygon) [reply .github/workflows/arm_deb_packager.yml]: Updated both the x86 and arm debian packagers with BUILD_SCRIPT_DISABLED=1 set for building the rust binary

**Comment by vcastellm** [comment]
Can you merge from develop once more? thx

**Pull request merged**